share/store/file.go

```diff
@@ -0,0 +1,21 @@
+package store
+
+import (
+	"context"
+	"io"
+
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+type File interface {
+	io.Closer
+	Size() int
+	ShareWithProof(ctx context.Context, axisType rsmt2d.Axis, axisIdx, shrIdx int) (share.Share, nmt.Proof, error)
```

Comment:
Suggestion:
Let's just call it `Share`. The WithProof part would be needed only if there is a version to get a share without a proof.